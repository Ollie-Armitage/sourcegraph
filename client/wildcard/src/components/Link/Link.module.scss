.anchor-link {
    color: var(--link-color);
    text-decoration: none;
    background-color: transparent; // Remove the gray background on active links in IE 10.

    &:hover {
        color: var(--link-hover-color);
        text-decoration: underline;
    }
}

// And undo these styles for placeholder links/named anchors (without href).
// It would be more straightforward to just use a[href] in previous block, but that
// causes specificity issues in many other styles that are too complex to fix.
// See https://github.com/twbs/bootstrap/issues/19402

.anchor-link:not([href]):not([class]) {
    &,
    &:hover {
        color: inherit;
        text-decoration: none;
    }
}
