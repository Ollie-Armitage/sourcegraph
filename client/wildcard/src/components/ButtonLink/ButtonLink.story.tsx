import { Meta } from '@storybook/react'
import React from 'react'

import { BrandedStory } from '@sourcegraph/branded/src/components/BrandedStory'
import webStyles from '@sourcegraph/web/src/SourcegraphWebApp.scss'

// TODO: To be update to wildcard Link component
import { AnchorLink, setLinkComponent } from './Link'

import { ButtonLink } from '.'

const Story: Meta = {
    title: 'wildcard/ButtonLink',

    decorators: [
        story => (
            <BrandedStory styles={webStyles}>{() => <div className="container mt-3">{story()}</div>}</BrandedStory>
        ),
    ],

    parameters: {
        component: ButtonLink,
        design: {
            type: 'figma',
            name: 'Figma',
            url: 'https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1',
        },
    },
}

export default Story
setLinkComponent(AnchorLink)

export const Simple = () => (
    <>
        <h3>Simple button link</h3>
        <ButtonLink to="#">Click me!</ButtonLink>

        <h3>Disabled button link</h3>
        <ButtonLink className="btn" disabled={true}>
            Disabled button
        </ButtonLink>
        <h3>Button link with undefined to property</h3>
        <ButtonLink>Click me</ButtonLink>
        <h3>Button link with onSelect</h3>
        <ButtonLink onSelect={() => alert('Link button selected')}>Click me</ButtonLink>
    </>
)
