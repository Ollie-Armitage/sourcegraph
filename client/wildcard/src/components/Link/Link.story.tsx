import { DecoratorFn, Meta, Story } from '@storybook/react'
import React from 'react'

import { BrandedStory } from '@sourcegraph/branded/src/components/BrandedStory'
import webStyles from '@sourcegraph/web/src/SourcegraphWebApp.scss'

import { Link, setLinkComponent, AnchorLink } from './Link'

const decorator: DecoratorFn = story => (
    <BrandedStory styles={webStyles}>{() => <div className="container mt-3">{story()}</div>}</BrandedStory>
)

const config: Meta = {
    title: 'wildcard/Link',

    decorators: [decorator],

    parameters: {
        component: Link,
        design: {
            type: 'figma',
            name: 'Figma',
            url: 'https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1',
        },
    },
}

export default config

setLinkComponent(AnchorLink)

export const LinkExample: Story = () => <Link to="#">Link to docs</Link>
