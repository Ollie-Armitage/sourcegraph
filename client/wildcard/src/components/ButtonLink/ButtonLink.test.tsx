import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import sinon from 'sinon'

import { ButtonLink } from './ButtonLink'
import { AnchorLink, setLinkComponent } from './Link'

describe('Button link', () => {
    test('renders correctly', () => {
        setLinkComponent(AnchorLink)
        const { asFragment } = render(<ButtonLink to="http://example.com">Button link</ButtonLink>)
        expect(asFragment()).toMatchSnapshot()
    })
    test('Works when to is undefined', () => {
        setLinkComponent(AnchorLink)
        const { container } = render(<ButtonLink>I am a link</ButtonLink>)
        expect(container.firstChild).toMatchInlineSnapshot(`
            <a
              class="nav-link"
              href=""
              role="button"
              tabindex="0"
            >
              I am a link
            </a>
        `)
    })
    test('Works with onSelect', () => {
        const handleClick = sinon.spy()
        setLinkComponent(AnchorLink)
        render(<ButtonLink onSelect={handleClick}>Click Me</ButtonLink>)
        fireEvent.click(screen.getByText(/click me/i))
        sinon.assert.called(handleClick)
    })
    test('Should be disabled with disabled prop', () => {
        render(
            <ButtonLink className="btn" disabled={true}>
                Disabled button
            </ButtonLink>
        )
        expect(screen.getByText(/disabled button/i)).toHaveClass('disabled')
    })
})
