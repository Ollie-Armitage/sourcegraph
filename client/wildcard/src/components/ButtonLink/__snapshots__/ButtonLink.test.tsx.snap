// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button link renders correctly 1`] = `
<DocumentFragment>
  <a
    class="nav-link"
    href="http://example.com"
    tabindex="0"
  >
     Button link 
  </a>
</DocumentFragment>
`;
